src/transformers/models/prophetnet/modeling_prophetnet.py

```diff
@@ -1686,7 +1686,7 @@ def prepare_attention_mask(self, hidden_states, attention_mask):
         batch_size, seq_length = hidden_states.shape[:2]
 
         # get causal mask
-        causal_mask = hidden_states.new(seq_length, seq_length).float().fill_(-float("inf"))
+        causal_mask = torch.full((seq_length, seq_length), -float("inf"), dtype=torch.float)
```

Comment:
We should also pass `device=hidden_states.device` here

Reply:
done